libass/ass_parse.c

```diff
@@ -512,11 +521,16 @@ char *parse_tags(ASS_Renderer *render_priv, char *p, char *end, double pwr,
             if (nargs && strncmp(start, "0", args->end - start)) {
                 skip_spaces(&start);
                 family = strndup(start, args->end - start);
-            } else
+            } else {
                 family = strdup(render_priv->state.style->FontName);
-            free(render_priv->state.family);
-            render_priv->state.family = family;
-            update_font(render_priv);
+            }
+            if (family) {
+                free(render_priv->state.family);
+                render_priv->state.family = family;
+                update_font(render_priv);
+            } else {
+                ass_nomem_log(render_priv->library, false, "Failed to strdup inline font!");
```

Comment:
Also not sure about keeping an old value in the case of a failure.

As a note: maybe we should introduce a hard limit for such values and use a fixed size array for it? It's a measure against malicious scripts with too long values and also it makes an allocation failure impossible.

Reply:
I don’t think a fixed-size array makes sense. This can’t grow indefinitely: at most, we duplicate almost the whole event text. Event text can be arbitrarily long, so it isn’t clear to me how imposing fixed length limits for individual parts of it will help. Of course, ideally we should be using pointer+length to begin with.

As for keeping the old value, again, the choice is between doing that and discarding the whole thing. Either pick is bad.